File: 2015/day_01/julia/attempt_2.jl
function main()
    s = chomp(read("../input.txt", String))
    level = 0
    position = 0

    for (i, c) in enumerate(s)
        level += (c == '(') ? 1 : (c == ')' ? -1 : 0)
        if position == 0 && level == -1
            position = i
        end
    end

    println(level, " ", position)
end

main()

File: 2015/day_01/rust/attempt_2.rs
const INPUT: &str = include_str!("../input.txt");

fn main() {
    let final_floor: i32 = INPUT.chars().map(|c| match c {
        '(' => 1,
        ')' => -1,
        _ => 0,
    }).sum();

    println!("Part 1: Final floor = {}", final_floor);

    let mut floor = 0;
    let mut position = None;

    for (i, c) in INPUT.chars().enumerate() {
        match c {
            '(' => floor += 1,
            ')' => floor -= 1,
            _ => {}
        }
        if floor == -1 {
            position = Some(i + 1);
            break;
        }
    }

    println!("Part 2: First basement position = {}", position.unwrap());
}
